src/Mod/Sketcher/Gui/TaskSketcherTool.cpp

```diff
@@ -0,0 +1,967 @@
+{
+    if (i == 0) {
+        ui->label->setVisible(val);
+        ui->parameterOne->setVisible(val);
+    }
+    else if (i == 1) {
+        ui->label2->setVisible(val);
+        ui->parameterTwo->setVisible(val);
+    }
+    else if (i == 2) {
+        ui->label3->setVisible(val);
+        ui->parameterThree->setVisible(val);
+    }
+    else if (i == 3) {
+        ui->label4->setVisible(val);
+        ui->parameterFour->setVisible(val);
+    }
+    else if (i == 4) {
+        ui->label5->setVisible(val);
+        ui->parameterFive->setVisible(val);
+    }
+    else if (i == 5) {
+        ui->label6->setVisible(val);
+        ui->parameterSix->setVisible(val);
+    }
+    else if (i == 6) {
+        ui->checkBoxPreserveFilletChamferCorner->setVisible(val);
+    }
+    else if (i == 7) {
+        ui->checkBoxTS2->setVisible(val);
+    }
+}
+
+void SketcherToolWidget::setParameterFocus(int i)
+{
+    if (i == 0) {
+        ui->parameterOne->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterOne, "setFocus", Qt::QueuedConnection);
+    }
+    else if (i == 1) {
+        ui->parameterTwo->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterTwo, "setFocus", Qt::QueuedConnection);
+    }
+    else if (i == 2) {
+        ui->parameterThree->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterThree, "setFocus", Qt::QueuedConnection);
+    }
+    else if (i == 3) {
+        ui->parameterFour->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterFour, "setFocus", Qt::QueuedConnection);
+    }
+    else if (i == 4) {
+        ui->parameterFive->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterFive, "setFocus", Qt::QueuedConnection);
+    }
+    else if (i == 5) {
+        ui->parameterSix->selectNumber();
+        QMetaObject::invokeMethod(ui->parameterSix, "setFocus", Qt::QueuedConnection);
+    }
+}
+
+void SketcherToolWidget::changeEvent(QEvent *e)
+{
+    QWidget::changeEvent(e);
+    if (e->type() == QEvent::LanguageChange) {
+        ui->retranslateUi(this);
+    }
+}
+
+// ----------------------------------------------------------------------------
+
+TaskSketcherTool::TaskSketcherTool(ViewProviderSketch *sketchView)
+    : TaskBox(Gui::BitmapFactory().pixmap("document-new"),tr("Tool settings"),true, 0)
+    , sketchView(sketchView)
+{
+    widget = new SketcherToolWidget(this, sketchView);
+    this->groupLayout()->addWidget(widget);
+
+    widget->loadSettings();
+
+    Gui::Selection().Attach(this);
+
+    Gui::Application* app = Gui::Application::Instance;
+    changedSketchView = app->signalChangedObject.connect(boost::bind
+        (&TaskSketcherTool::onChangedSketchView, this, bp::_1, bp::_2));
+    widget->setSettings(0);
+}
+
+TaskSketcherTool::~TaskSketcherTool()
+{
+    Gui::Selection().Detach(this);
+    //Content.erase(Content.begin());
+}
+
+void TaskSketcherTool::onChangedSketchView(const Gui::ViewProvider& vp,

```

Comment:
I am unsure if you have a planned use for this. If you do not, then we need to clean up the code.

It also happens to the framework I shared. There are headers that simply are not needed in some files.

Reply:
I don't know what it does and just left if after copy/paste the other task widget class.

Reply:
> @PaddleStroke
> 
> Do not worry. Most likely I won't have time to review before the next weekend. We will cross the bridge when we get there.

See here the updated version. It is now working but with several issues: https://github.com/PaddleStroke/FreeCAD/tree/paddle_widget